src/operator/smooth_l1_unary-inl.h

```diff
@@ -104,7 +104,7 @@ void SmoothL1BackwardUseIn_(const OutputGrad& out_grad,
 }
 
 MXNET_REGISTER_SIMPLE_OP(smooth_l1, XPU)
-.set_function(XPU::kDevMask, SmoothL1Forward_<XPU>, kInplaceInOut)
+.set_function(XPU::kDevMask, SmoothL1Forward_<XPU>, kNoInplace)
```

Comment:
We have DeclareBackwardDependency to ensure the in_data does not get
replaced. Previous version worked correctly thanks to this. I would say
Runtime Engine is quite robust. Let us change it now.
On Sun, May 15, 2016 at 12:43 PM Eric Junyuan Xie <notifications@github.com>
wrote:

> In src/operator/smooth_l1_unary-inl.h
> <https://github.com/dmlc/mxnet/pull/2136#discussion_r63286086>:
>
> > @@ -104,7 +104,7 @@ void SmoothL1BackwardUseIn_(const OutputGrad& out_grad,
> >  }
> >
> >  MXNET_REGISTER_SIMPLE_OP(smooth_l1, XPU)
> > -.set_function(XPU::kDevMask, SmoothL1Forward_<XPU>, kInplaceInOut)
> > +.set_function(XPU::kDevMask, SmoothL1Forward_<XPU>, kNoInplace)
>
> I just noticed this. Is this correct? Was the previous version buggy?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/dmlc/mxnet/pull/2136/files/403bd1655447110789ff1d6f4efa91e3d72f4d63#r63286086>
>